### PR TITLE
[1LP][RFR] Add negative test (wrong API port for container provider)

### PR DESCRIPTION
**Pull request by shalomnaim1**
{{ pytest : cfme/tests/containers/test_ssl_providers.py::test_setup_with_wrong_port --use-provider ocp-39-hawk }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit f4d8d118e9e911e7f8609f5cb0cd5239a8d8b874:

`cfme/tests/containers/test_ssl_providers.py`:
- :bangbang: Line 152:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 169:13: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 169:101: **E501** *line too long (108 > 100 characters)*
- :heavy_exclamation_mark: Line 173:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit fa5503059035585470c1a1df103048e987b91121:

`cfme/tests/containers/test_ssl_providers.py`:
- :bangbang: Line 21:52: **E231** *missing whitespace after ','*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by rlbabyuk** [comment]
@shalomnaim1 please check lint, otherwise looks good

**Comment by shalomnaim1** [comment]
@rlbabyuk Well now that's ok, I think you can approve :-)

File: cfme/tests/containers/test_ssl_providers.py [hunk 1]
```diff
@@ -148,3 +150,29 @@ def test_add_mertics_provider_ssl(provider, appliance, test_item,
...
+        node = new_provider.endpoints["default"].hostname
+        port = new_provider.endpoints["default"].api_port
+        assert (
+            'Credential validation was not successful: Failed to open TCP connection'
+            ' to {node}:{port} (No route to host - connect(2) for "{node}" port '
+            '{port}'.format(node=node, port=port) in ex.message), (
+            "No clear failure massage was provided for wrong api port")
+        is_fail = True
+
+    assert is_fail, "Provider was set with wrong api port"
```
Comment (quarckster) [inline cfme/tests/containers/test_ssl_providers.py]: **REQUIRED**
Please use `pytest.raises` context manager:
```python
with pytest.raises(SomeExceptionHere):
    new_provider.setup()
```
```python

Comment (shalomnaim1) [reply cfme/tests/containers/test_ssl_providers.py]: @quarckster Could you please explain what is the advantage of this code over my suggestion?

**Comment by dajoRH** [comment]
Lint report for commit cff74fc6afe721e34ebcea8d0763944ce2b926b4:

`cfme/tests/containers/test_ssl_providers.py`:
- :bangbang: Line 154:1: **E302** *expected 2 blank lines, found 1*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c64fd9be3dbcf3c1c5629c5cc68eb85c70f5ae3c:

`cfme/tests/containers/test_ssl_providers.py`:
- :bangbang: Line 169:25: **E127** *continuation line over-indented for visual indent*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Pull request merged**